src/cargo/core/compiler/mod.rs

```diff
@@ -1296,27 +1298,30 @@ fn on_stderr_line_inner(
                 message: String,
                 level: String,
             }
-            if let Ok(mut error) = serde_json::from_str::<CompilerMessage>(compiler_message.get()) {
-                if error.level == "error" && error.message.starts_with("aborting due to") {
+            if let Ok(mut msg) = serde_json::from_str::<CompilerMessage>(compiler_message.get()) {
+                if msg.message.starts_with("aborting due to")
+                    || msg.message.ends_with("warning emitted")
+                    || msg.message.ends_with("warnings emitted")
+                {
                     // Skip this line; we'll print our own summary at the end.
                     return Ok(true);
                 }
                 // state.stderr will add a newline
-                if error.rendered.ends_with('\n') {
-                    error.rendered.pop();
+                if msg.rendered.ends_with('\n') {
+                    msg.rendered.pop();
                 }
                 let rendered = if options.color {
-                    error.rendered
+                    msg.rendered
                 } else {
                     // Strip only fails if the the Writer fails, which is Cursor
                     // on a Vec, which should never fail.
-                    strip_ansi_escapes::strip(&error.rendered)
+                    strip_ansi_escapes::strip(&msg.rendered)
                         .map(|v| String::from_utf8(v).expect("utf8"))
                         .expect("strip should never fail")
                 };
                 if options.show_warnings {
```

Comment:
I'm a little confused about this looking at this again although this isn't an issue introduced by this PR. 

Despite its name it appears that this option is controlling both errors and warnings, and I was originally worried that this could sweep errors under the rug. It looks like, though, that this is only set to `false` when replaying back caches, which if we have a cache we're guaranteed there's no errors.

Perhaps this could instead be rename to `show_diagnostics` or similar?

Reply:
Sure!  I had renamed it earlier, but changed it back when I went with a different approach.  I found it confusing, too.  I think it was originally `show_warnings` since it only applied to cache replay which doesn't record errors, so warnings was the only thing that mattered.  I added a comment to try to explain what it means.